src/components/post/index.js

```diff
@@ -4,33 +4,39 @@ import Comment from "../comment"
 import OptionsIcon from "../optionsIcon"
 import EditPostModal from "../editPostModal"
 import ProfileCircle from "../profileCircle"
-import { useEffect, useState } from "react"
+import { useState, useEffect, useContext } from "react"
+import { toggleLike } from "../../service/apiClient"
 import "./style.css"
 
 import emptyHeart from "../../assets/icons/empty-heart.png"
 import heart from "../../assets/icons/heart.png"
 import emptyComment from "../../assets/icons/empty-comment.png"
 import comment from "../../assets/icons/comment.png"
+import jwtDecode from "jwt-decode"
 import CommentInput from "../commentInput"
+import { AuthContext } from "../../context/auth"
 
 const Post = ({
   postId,
   name,
   date,
   content,
   comments = [],
-  likes = 0,
+  likes,
   getAllPosts,
 }) => {
   const { openModal, setModal } = useModal()
   const [postContent, setPostContent] = useState(null)
-  const [isLike, setIsLike] = useState(false)
+  const [userLiked, setUserLiked] = useState(false)
+  const [likesCount, setLikesCount] = useState(likes.length)
   const [isComment, setIsComment] = useState(false)
 
   const [formatDate, setFormatDate] = useState(null)
 
   const userInitials = name.match(/\b(\w)/g)
 
+  const { token } = useContext(AuthContext)
```

Comment:
You can just use `useAuth`
https://github.com/boolean-uk/team-dev-client-c11/blob/e68c4414dc4c7d60057c29185d524d04fa6c88b9/src/pages/register/index.js#L9